examples/gno.land/p/moul/xmath/generator.go

```diff
@@ -0,0 +1,183 @@
+//go:build ignore
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+type Type struct {
+	Name      string
+	ZeroValue string
+	Signed    bool
+	Float     bool
+}
+
+var types = []Type{
+	{"Int8", "0", true, false},
+	{"Int16", "0", true, false},
+	{"Int32", "0", true, false},
+	{"Int64", "0", true, false},
+	{"Int", "0", true, false},
+	{"Uint8", "0", false, false},
+	{"Uint16", "0", false, false},
+	{"Uint32", "0", false, false},
+	{"Uint64", "0", false, false},
+	{"Uint", "0", false, false},
+	{"Float32", "0.0", true, true},
+	{"Float64", "0.0", true, true},
+}
+
+const sourceTpl = `package xmath
```

Comment:
I would add a comment like `// Code generated by generator.go; DO NOT EDIT.`; see `go help generate`

```
To convey to humans and machine tools that code is generated,
generated source should have a line that matches the following
regular expression (in Go syntax):

	^// Code generated .* DO NOT EDIT\.$

This line must appear before the first non-comment, non-blank
text in the file.
```

Reply:
5b91430b